### PR TITLE
Add resolver for tracking first required AT Version

### PR DESCRIPTION
**Pull request by howard-e**
This PR adds support for tracking **"First Required AT Versions** as outlined in #792.

* Includes `firstRequiredAtVersion` on graphql schema's `TestPlanVersion`. When provided an AT id, given that a report has been generated for that test plan version with that AT, it will follow defined rules to return the first required AT version for a TestPlanVersion once it has gotten to RECOMMENDED.

Since this is work is also dependent on a "primary test plan run", this is branched off of the work being done on #1001.

### CODE REVIEW

**Comment by alflennik** [comment]
Cannot read properties of undefined while querying the new field.

I got this error with the query
```
query { 
  testPlanVersions {
    id
    title
    versionString
    firstRequiredAtVersion(atId: 1) {
 			id
    }
  }
}
```

Any idea what might be causing this?

This is with a fresh database and the sample data.

File: server/graphql-schema.js [hunk 1]
```diff
@@ -377,6 +377,18 @@ const graphqlSchema = gql`
...
+        for this TestPlanVersion.
+
+        A first required AT Version is the earliest version used to collect
+        results for a test plan run marked as primary, and is a required report.
+
+        This should be used to prevent any older AT Versions from being used to
+        create subsequent reports after the Test Plan Version has reached
+        RECOMMENDED.
+        """
+        firstRequiredAtVersion(atId: ID!): AtVersion
```
Comment (alflennik) [inline server/graphql-schema.js:391]: I was wondering, how do you feel about calling this `earliestMissingAtVersion`? I feel like "missing" might be more descriptive than "required", possibly?

File: server/graphql-schema.js [hunk 2]
```diff
@@ -377,6 +377,18 @@ const graphqlSchema = gql`
...
+        """
+        The first required AT Version of a specified AT used to collect results
+        for this TestPlanVersion.
+
+        A first required AT Version is the earliest version used to collect
+        results for a test plan run marked as primary, and is a required report.
+
+        This should be used to prevent any older AT Versions from being used to
+        create subsequent reports after the Test Plan Version has reached
+        RECOMMENDED.
```
Comment (alflennik) [inline server/graphql-schema.js:389]: I appreciate the depth of this explanation, but even so I had a hard time understanding what this field really means. Would it be possible to describe it in terms of how exactly it fits into the workflow for test plans?

**Review by alflennik — CHANGES_REQUESTED** [review]
Just a couple of notes! The main thing is that I encountered an error in the graphql playground.

**Comment by howard-e** [comment]
> Cannot read properties of undefined while querying the new field.
> I got this error with the query
> 
> ```
> query { 
>   testPlanVersions {
>     id
>     title
>     versionString
>     firstRequiredAtVersion(atId: 1) {
>  			id
>     }
>   }
> }
> ```
> 
> Any idea what might be causing this?
> 
> This is with a fresh database and the sample data.

https://github.com/w3c/aria-at-app/pull/1051/commits/21a0c72ac25921d84de96cee99ad3125a41d3cfa should fix this. Because of how the `testPlanVersionsResolver` is done the, the `testPlanReports` and `phase` variables were being left out which would then throw the error.

File: server/graphql-schema.js (hunk 2 above)
Comment (howard-e) [reply server/graphql-schema.js]: Updated in https://github.com/w3c/aria-at-app/pull/1051/commits/b916cafdc60e66919b9306991eee53cf1e283970. Does that help to clarify?

File: server/graphql-schema.js (hunk 1 above)
Comment (alflennik) [reply server/graphql-schema.js]: Thanks!

**Pull request merged**